crates/ide/src/goto_implementation.rs

```diff
@@ -337,6 +337,42 @@ impl Tr for S {
     const C: usize = 4;
         //^
 }
+"#,
+        );
+    }
+
+    #[test]
+    fn goto_implementation_in_nested_blocks_1() {
+        check(
+            r#"
+//- minicore: copy, derive
+fn test() {
+    #[derive(Copy)]
+  //^^^^^^^^^^^^^^^
+    struct Foo$0;
+}
+"#,
+        );
+    }
+
+    #[test]
+    fn goto_implementation_in_nested_blocks_2() {
```

Comment:
This test we cannot support due to performance reasons, see the main review message